bjnp-commands.c:188:9: error: comparison of array 'printer_id' not equal to a null pointer is always true [-Werror,-Wtautological-pointer-compare]
    if (printer_id != NULL) {
        ^~~~~~~~~~    ~~~~

Index: bjnp-commands.c
--- bjnp-commands.c.orig
+++ bjnp-commands.c
@@ -185,10 +185,8 @@ get_printer_id(http_addr_t *addr, char *model, char *I
 
     /* set IEEE1284_id */
 
-    if (printer_id != NULL) {
-        strncpy(printer_id, id.udp_identity_response.id, id_len);
-        printer_id[id_len] = '\0';
-    }
+    strncpy(printer_id, id.udp_identity_response.id, id_len);
+    printer_id[id_len] = '\0';
 
     bjnp_debug(LOG_INFO, "Identity = %s\n", printer_id);
 
